javalib/src/main/scala/java/lang/NumberParser.scala

```diff
@@ -0,0 +1,246 @@
+package java.lang
+
+private[lang] object NumberParser {
+  private val FLOAT_MIN_EXP  = -46
+  private val FLOAT_MAX_EXP  = 38
+  private val DOUBLE_MIN_EXP = -324
+  private val DOUBLE_MAX_EXP = 308
+
+  private case class StringExponentPair(s: _String,
+                                        e: scala.Int,
+                                        negative: scala.Boolean)
+
+  @native def parseDblImpl(s: _String, e: scala.Int): scala.Double
+  @native def parseFltImpl(s: _String, e: scala.Int): scala.Float
```

Comment:
These need to be reimplemented in Scala. Native methods are implemented in `C/C++` parts of Apache Harmony.